test/groupBy.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('groupBy', function() {
+
+  eq(typeof S.groupBy, 'function');
+  eq(S.groupBy.length, 2);
+  eq(S.groupBy.toString(), 'groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)');
+
+  var productsOf3 = S.curry2(function(a, b) {
+    return a * b % 3 === 0;
+  });
+
+  eq(S.groupBy(productsOf3, []), []);
+  eq(S.groupBy(productsOf3, [[]]), [[[]]]);
```

Comment:
I noticed that too. But in Sanctuary that would be caught by defining `productsOf3` with `def`. No way for `groupBy` to know whether it's passing an invalid argument into `f`.